modules/aws-vpn-vpnendpoint/main.tf

```diff
@@ -65,7 +65,7 @@ resource "aws_ec2_client_vpn_network_association" "my-vpn_sso" {
 
   client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.my-vpn_sso.id
   subnet_id              = each.value
-  security_groups        = [aws_security_group.my-vpn.id]
+  security_group_ids     = [aws_security_group.my-vpn.id]
```

Comment:
it seems the right place to put this option is in `aws_ec2_client_vpn_endpoint` resource and remove security_groups from here, check docs here:
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_client_vpn_network_association
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ec2_client_vpn_endpoint